src/Entity/Incident/IncidentRepository.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace AppBundle\Entity\Incident;
+
+use AppBundle\Entity\Sylius\OrderVendor;
+use Doctrine\ORM\EntityRepository;
+/**
+ * @extends EntityRepository<Incident>
+ */
+class IncidentRepository extends EntityRepository {
+
+
+    //TODO: Move this function in a helper class
+    private function transformArray(array $inputArray): array {
+        $outputArray = [];
+
+        foreach ($inputArray as $key => $value) {
+            $keys = explode("_", $key);
+            /** @var array $tempArray */
+            $tempArray = &$outputArray;
+
+            foreach ($keys as $nestedKey) {
+                if (!isset($tempArray[$nestedKey])) {
+                    $tempArray[$nestedKey] = [];
+                }
+                $tempArray = &$tempArray[$nestedKey];
+            }
+            $tempArray = is_array($value) ? $this->transformArray($value) : $value;
+        }
+
+        return $outputArray;
+    }
+
+    public function getAllIncidents(): array {
```

Comment:
At some point if there are lots of incidents, this will die. We should have query limit / pagination here.

Reply:
Yep i'm aware of that problem but i guess it's okay for now